### PR TITLE
Deploy redis

### PR DESCRIPTION
**Pull request by temitope777**
**Context**  
Deploy Redis

**Changes proposed in this pull request**  
Add redis resource in aks using terraform-modules  

**Guidance to review**
make development_aks terraform-apply-aks

### CODE REVIEW

File: terraform/aks/database.tf [hunk 1]
```diff
@@ -15,3 +15,20 @@ module "postgres" {
...
+  namespace             = var.namespace
+  environment           = var.environment
+  azure_resource_prefix = var.azure_resource_prefix
+  service_short         = var.service_short
+  config_short          = var.config_short
+  service_name          = var.service_name
+
+  cluster_configuration_map = module.cluster_data.configuration_map
+
+  use_azure               = true
```
Comment (saliceti) [inline terraform/aks/database.tf:31]: Can we use var.deploy_azure_backing_services here?

File: terraform/aks/database.tf [hunk 2]
```diff
@@ -15,3 +15,20 @@ module "postgres" {
...
+  azure_enable_monitoring = var.enable_monitoring
+  azure_patch_schedule    = [{ "day_of_week": "Sunday", "start_hour_utc": 01 }]
```
Comment (divesh987) [inline terraform/aks/database.tf:33]:
```suggestion
  azure_patch_schedule    = [{ "day_of_week": "Sunday", "start_hour_utc": 01 }]
  azure_maxmemory_policy  = "allkeys-lfu"
```
The schools experience PAAS application redis instance uses allkeys-lfu as its maxmemory policy as shown here https://github.com/DFE-Digital/schools-experience/blob/master/terraform/paas/services.tf line 53 which differs from the terraform-modules default which is allkeys-lru so we need this variable to updated here

File: terraform/aks/database.tf [hunk 3]
```diff
@@ -15,3 +15,20 @@ module "postgres" {
...
+  environment           = var.environment
+  azure_resource_prefix = var.azure_resource_prefix
+  service_short         = var.service_short
+  config_short          = var.config_short
+  service_name          = var.service_name
+
+  cluster_configuration_map = module.cluster_data.configuration_map
+
+  use_azure               = true
+  azure_enable_monitoring = var.enable_monitoring
```
[the remark this answers is not in the document]
Comment (temitope777) [reply terraform/aks/database.tf:32]: done

File: terraform/aks/database.tf (hunk 2 above)
Comment (temitope777) [reply terraform/aks/database.tf]: added

**Review by saliceti — APPROVED** [review]

**Pull request merged**